File: solutions/2434.py
# -*- coding: utf-8 -*-
_input = input()
n, balance = _input.split()
n = int(n)
balance = int(balance)
worst = balance

while n:
    n -= 1
    value = input()
    balance += int(value)
    worst = balance if balance < worst else worst

print(worst)

File: solutions/2434.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n').slice(0, -1);

let [n, balance] = lines[0].split(' ');
let worst = parseInt(balance);

balance = parseInt(balance);
lines = lines.slice(1);
lines.forEach(value => {
  balance += parseInt(value);
  worst = balance < worst ? balance : worst;
});
console.log(worst);
